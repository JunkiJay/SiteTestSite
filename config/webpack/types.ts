export type BuildMode = 'development' | 'production';

export interface BuildOptions {
  mode: BuildMode;
  devServerPort: string;
  paths: {
    entry: string;
    dist: string;
    public: string;
    src: string;
    root: string;
  };
  analyzer?: 'true' | 'false';
}
