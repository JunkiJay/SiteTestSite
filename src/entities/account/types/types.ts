import { Company, Member } from '@entities/company/types';

export interface AccountState {
  currentCompany: Company | null;
  friends: Member[];
  ownerCompanyId: string | null;
}

export interface FetchAccountDataResponse {
  currentCompany: Company | null;
  friends: Member[];
  ownerCompanyId: string | null;
}
