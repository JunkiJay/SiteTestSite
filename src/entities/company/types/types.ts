export interface Member {
  login: string;
  id: string;
  image?: string;
}

export interface Company {
  id: string;
  image: string;
  name: string;
  field: string;
  earned: number;
  members: Member[];
}

export interface CompanyFormValues {
  image: File | string;
  name: string;
  field: string;
}
