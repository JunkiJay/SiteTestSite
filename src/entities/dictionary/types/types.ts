export interface EarnEvent {
  id: string;
  title: string;
  reward: string;
  link: string;
  isCompleted: boolean;
}
