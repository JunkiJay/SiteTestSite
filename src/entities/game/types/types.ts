export type Rank = 'boss' | 'student';

export enum EBoostType {
  rechargingSpeed = 'rechargingSpeed',
  multiTap = 'multiTap',
  maxEnergy = 'maxEnergy',
  autoClicker = 'autoClicker',
}

export interface Boost {
  price: number;
  level: number;
  type: EBoostType;
}

export type GameBoosts = Record<keyof typeof EBoostType, Boost>;

export enum EGameEventType {
  tap = 'tap',
  regen = 'regen',
  autoClick = 'autoClick',
}

export interface GameEvent {
  type: EGameEventType;
  id: number;
}

export interface GameSliceState {
  energy: number;
  capital: number;
  gameEvents: GameEvent[];
  boosts: GameBoosts;
  lastSyncId: number;
  rank: Rank;
}

export interface SyncMutationPayload {
  [EGameEventType.tap]: number;
  [EGameEventType.regen]: number;
  [EGameEventType.autoClick]: number;
  id: number;
}

export interface SyncMutationResponse {
  energy: number;
  capital: number;
  id: number;
}

export interface FetchGameDataResponse {
  energy: number;
  capital: number;
  boosts: Boost[];
  rank: Rank;
}

export interface ClaimRewardPayload {
  id: string;
  reward: number;
}
