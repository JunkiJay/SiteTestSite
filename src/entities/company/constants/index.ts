export * from './defailt-company-form-values';
export * from './validation-schema';
