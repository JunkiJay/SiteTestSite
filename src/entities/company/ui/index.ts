export * from './companies-list';
export * from './company-fields';
export * from './owner-company';
export * from './participant-company';
