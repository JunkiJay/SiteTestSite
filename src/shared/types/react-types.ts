import { ReactNode } from 'react';

export interface ComponentWithChildren<T = ReactNode> {
  children: T;
}
