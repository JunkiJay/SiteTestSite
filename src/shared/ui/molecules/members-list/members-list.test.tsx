import { render, screen } from '@testing-library/react';
import { MembersList } from '.';
import { Member } from '@entities/company/types';

describe('', () => {
  const members: Member[] = [
    { id: '1', image: '', login: 'Login 1' },
    { id: '2', image: '', login: 'Login 2' },
    { id: '3', image: '', login: 'Login 3' },
  ];

  it('should rendered', () => {
    render(<MembersList members={members} title="Members list" />);
    const listElement = screen.getByRole('list');
    expect(listElement).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<MembersList members={members} title="Members list" />);

    expect(container).toMatchSnapshot();
  });
});
