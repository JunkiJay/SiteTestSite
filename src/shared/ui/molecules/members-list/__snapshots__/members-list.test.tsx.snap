// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1e2sded-MuiPaper-root"
    role="list"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-z2eky3-MuiTypography-root"
    >
      Members list
    </p>
    <div
      class="MuiBox-root css-163g6ns"
    />
  </div>
</div>
`;
